src/sentry/explore/models.py

```diff
@@ -60,10 +60,13 @@ class ExploreSavedQuery(DefaultFieldsModel):
         choices=ExploreSavedQueryDataset.as_choices(), default=ExploreSavedQueryDataset.SPANS
     )
     is_multi_query = models.BooleanField(default=False)
+    prebuilt_id = BoundedPositiveIntegerField(null=True, db_default=None)
+    prebuilt_version = BoundedPositiveIntegerField(null=True, db_default=None)
 
     class Meta:
         app_label = "explore"
         db_table = "explore_exploresavedquery"
+        unique_together = (("organization", "prebuilt_id"),)
```

Comment:
I'm still reviewing the endpoint logic, but from what I understand this is saying that you'll share a single prebuilt across an org instead of per user. When a user stars it, they'll get their own version assigned to them that they can change and it'll be isolated from the prebuilt ones, correct?

Reply:
`explore_exploresavedquery` is an org level object. When a user stars a saved query, the relationship between `user_id` and `saved_query_id` is saved in a separate table. None of the query filters (including the `prebuilt_id`) are duplicated into new rows during the starring process. If a user wants to edit and save from a prebuilt query, they must save as a new query (`prebuilt_id` is not propagated in this process) so this constraint doesn't get violated here either.